examples/vision/sample_size_estimate.py

```diff
@@ -0,0 +1,479 @@
+
+    history = model.fit(
+        x=training_data,
+        y=training_labels,
+        batch_size=batch_size,
+        epochs=epochs,
+        validation_split=0.1,
+        callbacks=[stopper],
+    )
+    return history
+
+
+def unfreeze(model, block_name, verbose=0):
+    """Unfreezes Keras model layers.
+
+    Arguments:
+        model: Keras model.
+        block_name: Str, layer name for example block_name = 'block4'.
+                    Checks if supplied string is in the layer name.
+        verbose: Int, 0 means silent, 1 prints out layers trainability status.
+
+    Returns:
+        Keras model with all layers after (and including) the specified
+        block_name to trainable, excluding BatchNormalization layers.
+    """
+
+    # Unfreeze from block_name onwards
+    set_trainable = False
+
+    for layer in model.layers:
+        if block_name in layer.name:
+            set_trainable = True
+        if set_trainable and not isinstance(layer, layers.BatchNormalization):
+            layer.trainable = True
+            if verbose == 1:
+                print(layer.name, "trainable")
+        else:
+            if verbose == 1:
+                print(layer.name, "NOT trainable")
+    print("Trainable weights:", len(model.trainable_weights))
+    print("Non-trainable weights:", len(model.non_trainable_weights))
+    return model
+
+
+"""
+# Define iterative training function
+
+To train a model over several subsample sets we need to create an iterative training function.
+"""
+
+
+def train_model(training_data, training_labels):
+    """Trains the model as follows:
+    
+    - Trains only the top layers for 10 epochs.
+    - Unfreezes deeper layers.
+    - Train for 20 more epochs.
+
+    Arguments:
+        training_data: NumPy Array, trainig data.
+        training_labels: NumPy Array, trainig labels.
+
+    Returns:
+        Model accuracy.
+    """
+
+    model = build_model(num_classes)
+
+    # Compile and train top layers
+    history = compile_and_train(
+        model,
+        training_data,
+        training_labels,
+        metrics=[keras.metrics.AUC(name="auc"), "acc"],
+        optimizer=keras.optimizers.Adam(),
+        patience=3,
+        epochs=10,
+    )
+
+    # Unfreeze model from block 6 onwards
+    model = unfreeze(model, "block_10")
+
+    # Compile and train for 20 epochs with a lower learning rate
+    fine_tune_epochs = 20
+    total_epochs = history.epoch[-1] + fine_tune_epochs
+
+    history_fine = compile_and_train(
+        model,
+        training_data,
+        training_labels,
+        metrics=[keras.metrics.AUC(name="auc"), "acc"],
+        optimizer=keras.optimizers.Adam(learning_rate=1e-4),
+        patience=5,
+        epochs=total_epochs,
+    )
+
+    # Calculate model accuracy on the test set
+    y_pred = model.predict(img_test)
+    y_pred_indices = np.argmax(y_pred, axis=1)
+    acc = np.sum(np.equal(y_pred_indices, np.argmax(label_test, axis=1))) / len(
```

Comment:
Why not use `model.evaluate()`?

Reply:
Ah, yes thanks for the suggestion. Much more elegant.

Reply:
Thanks for the copyedits, much appreciated. Looking really good now. I've made a few minor edits and changed in model.evaluate().